src/pepr/operator/crd/sources/package/v1alpha1.ts

```diff
@@ -123,6 +123,12 @@ const allow = {
           Only valid for Ingress rules.",
         type: "string",
       },
+      serviceAccount: {
+        description:
+          "The service account to restrict outgoing traffic from within the package namespace. \
+          Only valid for Egress rules.",
```

Comment:
Based on this PR, isn't a more accurate description that we only apply the source `serviceAccount` for Egress rules with `remoteHost` targets?

We create [authpolicies](https://github.com/defenseunicorns/uds-core/blob/979d251e44e35784b86153e8df1c487cf3a6dc5c/src/pepr/operator/controllers/network/authorizationPolicies.ts#L164) for other egress rules but I don't think this PR adds the serviceAccount to that logic.  Maybe we should incorporate this field into that logic??  Thoughts here @mjnagel / @chance-coleman ?

Reply:
> We create [authpolicies](https://github.com/defenseunicorns/uds-core/blob/979d251e44e35784b86153e8df1c487cf3a6dc5c/src/pepr/operator/controllers/network/authorizationPolicies.ts#L164) for other egress rules but I don't think this PR adds the serviceAccount to that logic. Maybe we should incorporate this field into that logic??

I might be forgetting a different change, but current authpolicy logic is [entirely _Ingress_ focused](https://github.com/defenseunicorns/uds-core/blob/979d251e44e35784b86153e8df1c487cf3a6dc5c/src/pepr/operator/controllers/network/authorizationPolicies.ts#L225) I believe. The linked function for example is used to allow ingress from a remote service account -> a pod (with the destination being selected via labels rather than service account).

Reply:
```suggestion
          Only valid for Egress rules using `RemoteHosts`.",
```

I believe this would be a more accurate description right?  Technically we could have a egress rule without `RemoteHosts` and this field would do nothing?  Maybe it should do something in the future?

Example- Create an egress rule to `remoteNamespace`, as the logic is written, I think the authpolicy would not enforce the originating serviceAccount correct?

If I am understanding this correctly, I think we should
1. Clarify that serviceAccount is only applicable for egress rules with `remoteHost`
2. Maybe we should create a follow up issue to add `ServiceAccount` in our existing auth policy logic? Would defer to @mjnagel on if this is desirable.

Reply:
Yeah, I think maybe you're right and might be good to add an additional validation here to check if serviceAccount, that remoteHost is also defined, so it's clear they belong together? There is going to be an interesting intersection of logic for all those allow fields...

Reply:
Yeah a validation actually might be a good idea.  Block use of ServiceAccount unless using remote host.  And then maybe in the future if we wanted to extend the serviceaccount logic we could tie that into our other egress authpolicies

Reply:
added a validation and a test for it